geode-core/src/main/java/org/apache/geode/internal/cache/DirectoryHolder.java

```diff
@@ -46,20 +45,27 @@
   private final DiskDirectoryStats dirStats;
 
   /** For testing purposes we can set the disk directory size in bytes **/
-  @MutableForTesting
-  static DiskDirSizesUnit DISK_DIR_SIZES_UNIT = DiskDirSizesUnit.MEGABYTES;
+  DiskDirSizesUnit diskDirSizesUnit;
 
   DirectoryHolder(StatisticsFactory factory, File dir, long space, int index) {
     this(dir.getPath(), factory, dir, space, index);
   }
 
   DirectoryHolder(String ownersName, StatisticsFactory factory, File dir, long space, int index) {
+    this(ownersName, factory, dir, space, index, DiskDirSizesUnit.MEGABYTES);
+  }
+
+  DirectoryHolder(String ownersName, StatisticsFactory factory, File dir, long space, int index,
+      DiskDirSizesUnit unit) throws RuntimeException {
```

Comment:
The `throws RuntimeException` is not needed here because IllegalArgumentException is an unchecked exception. Also, consider adding the `@VisibleForTesting` annotation if this constructor will only be used for testing purposes.